Let Debian's sensible-editor guess the editor.
Index: conkeror-0.9~git090923/modules/external-editor.js
===================================================================
--- conkeror-0.9~git090923.orig/modules/external-editor.js	2009-09-23 00:36:26.000000000 +0200
+++ conkeror-0.9~git090923/modules/external-editor.js	2009-09-23 00:39:48.000000000 +0200
@@ -5,7 +5,7 @@
  * COPYING file.
 **/
 
-define_variable("editor_shell_command", getenv("VISUAL") || getenv("EDITOR") || "emacs",
+define_variable("editor_shell_command", "/usr/bin/sensible-editor",
     "Shell command used to invoke an external editor.\n" +
     "This defaults to the value of the EDITOR environment variable.  If " +
     "`run_external_editor_function' is non-null, it is used instead to " +
