Let Debian's sensible-editor guess the editor.
Index: conkeror/modules/external-editor.js
===================================================================
--- conkeror.orig/modules/external-editor.js	2012-01-31 18:28:37.000000000 +0100
+++ conkeror/modules/external-editor.js	2012-01-31 18:35:45.000000000 +0100
@@ -5,7 +5,7 @@
  * COPYING file.
 **/
 
-define_variable("editor_shell_command", getenv("VISUAL") || getenv("EDITOR") || "emacs",
+define_variable("editor_shell_command", "/usr/bin/sensible-editor",
     "Shell command used to invoke an external editor.\n" +
     "This defaults to the value of the EDITOR environment variable.  If " +
     "`run_external_editor_function' is non-null, it is used instead to " +
